### PR TITLE
Clarify the project's licences

### PR DESCRIPTION
**Pull request by kcgen**
In anticipation of adding the GPL-3.0-or-later [mverb library](https://github.com/martineastwood/mverb), this PR adopts the Inkscape project's detailed [COPYING](https://gitlab.com/inkscape/inkscape/-/blob/master/COPYING) terms.

Like us, Inkscape's own source files are licensed GPL-2.0-or-later, however the project as a whole makes use of LGPL, GPL, BSD, MIT, and Public Domain licensed code.

Even more timely for us is that Inkscape uses third-party GPL-3.0-or-later code, which we plan to do with mverb.

That said, [DOSBox SVN commit 3026](https://sourceforge.net/p/dosbox/code-0/3026#diff-4) already added GPL-3.0-or-later code, so from an overall license perspective, not much will change when we add mverb. But I liked Inkscape's approach, so why not try to clarify things a bit more! :slightly_smiling_face:

### CODE REVIEW

**Review by johnnovak — APPROVED** [review]
Thanks for taking care of this @kcgen 🙇🏻‍♂️ 

I'm a BSD/WTFPL/Public Domain type of person myself, but I get it that the legal stuff can matter in a project like this and needs to be sorted out (and I'm glad it's not me who had to do it 😅)

File: COPYING [hunk 1]
```diff
@@ -1,339 +1,85 @@
...
+- MAME's Philips SAA1099 and Texas Instruments SN76489 emulators,
+  under the src/hardware/mame directory, are both licensed under
+  the BSD 3-Clause terms.
 
-  To protect your rights, we need to make restrictions that forbid
-anyone to deny you these rights or to ask you to surrender the rights.
-These restrictions translate to certain responsibilities for you if you
-distribute copies of the software, or if you modify it.
+- MandrakeSoft's National Semiconductor NE2000 emulator, in file
+  src/hardware/ne2000.cppwhich, is licensed under the
```
Comment (kcgen) [inline COPYING:21]: `cppwhich` -> cpp, which

File: COPYING [hunk 2]
```diff
@@ -1,339 +1,85 @@
...
+  LGPL-2.1-or-later terms.
 
-  For example, if you distribute copies of such a program, whether
-gratis or for a fee, you must give the recipients all the rights that
-you have.  You must make sure that they, too, receive or can get the
-source code.  And you must show them these terms so they know their
-rights.
+- Timothy Lottes CRT-styled scan-line GLSL shader, in file
```
Comment (kcgen) [inline COPYING:24]: Timothy Lottes'

File: COPYING [hunk 3]
```diff
@@ -1,339 +1,85 @@
...
-on the Program, the distribution of the whole must be on the terms of
-this License, whose permissions for other licensees extend to the
-entire whole, and thus to each and every part regardless of who wrote it.
+ - LGPL-3.0-or-later -> GPL-3.0-or-later
 
-Thus, it is not the intent of this section to claim rights or contest
-your rights to work written entirely by you; rather, the intent is to
-exercise the right to control the distribution of derivative or
-collective works based on the Program.
+Taken as a whole, the compiled binaries of DOSBox Staging project
```
Comment (kcgen) [inline COPYING:73]: compiled DOSBox Staging binaries

File: COPYING [hunk 4]
```diff
@@ -1,339 +1,85 @@
...
+are covered by the terms of GNU GPL version 3 or later, based on
+GNU's license compatibility flow chart [1] and FAQ answers
+regarding compatibility [2].
```
Comment (kcgen) [inline COPYING:76]: Combine this section with the first para

File: licenses/GPL-2.0-or-later.txt [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+
+The license text can be found in GPL-2.0.txt.
```
Comment (shermp) [inline licenses/GPL-2.0-or-later.txt:14]: `GPL-2.0.txt` seems to be missing from this PR.

**Review by shermp — COMMENTED** [review]
Should `COPYING` be replaced with the new text in `LICENSE`?

Would make it easier to read the PR IMHO, and maybe a better fit?

**Comment by kcgen** [comment]
> Should COPYING be replaced with the new text in LICENSE?

Are we allowed to have a complex description of all the licenses inside a "singular" file call LICENSE?

My understand is that the LICENSE file should be "the license", where as COPYING describes how people can take/use/copy aspects of the code.

**Comment by shermp** [comment]
> > Should COPYING be replaced with the new text in LICENSE?
> 
> Are we allowed to have a complex description of all the licenses inside a "singular" file call LICENSE?
> 
> My understand is that the LICENSE file should be "the license", where as COPYING describes how people can take/use/copy aspects of the code.

Hmm, actually, not sure on that.

**Comment by kcgen** [comment]
Maybe we could have some description in LICENSE saying that this is a multi-licensed project, and that details are provided in COPYING? 

(I noticed Inkscape doesn't have a LICENSE either: https://gitlab.com/inkscape/inkscape/-/tree/master)

File: licenses/GPL-2.0-or-later.txt (hunk 5 above)
Comment (kcgen) [reply licenses/GPL-2.0-or-later.txt:14]: Will add - thanks @shermp (keeping unresolved until added).

**Comment by GranMinigun** [comment]
QEMU describes its different licenses in the LICENSE file and places the project's primary license in COPYING.

**Comment by kcgen** [comment]
> QEMU describes its different licenses in the LICENSE file and places the project's primary license in COPYING.

Do we want to go with that?
 - I would move `COPYING` over to `LICENSE`
 - Then make COPYING basically a link (or refer) to the `GPL-2.0-or-later.txt`

(I'm happy with this approach - it seems more complete).

File: COPYING [hunk 6]
```diff
@@ -1,339 +1,85 @@
...
 
-  We protect your rights with two steps: (1) copyright the software, and
-(2) offer you this license which gives you legal permission to copy,
-distribute and/or modify the software.
+- Libraries in src/libs directory, as licensed as follows:
+   - decoders/archive.h, stb.h, and stb_vorbis.h: Public Domain
+   - decoders/dr_flac.h, dr_mp3.h, and dr_wav.h: Public Domain or MIT
+   - decoders/xxhash.h: BSD 2-Clause
+   - enet: MIT
+   - ghc: Public Domain-compatible
```
Comment (kcgen) [inline COPYING:33]: > ghc::filesystem is MIT

Thanks @shermp.

File: COPYING [hunk 7]
```diff
@@ -1,339 +1,85 @@
...
+  LGPL-2.1-or-later terms.
 
-  For example, if you distribute copies of such a program, whether
-gratis or for a fee, you must give the recipients all the rights that
-you have.  You must make sure that they, too, receive or can get the
-source code.  And you must show them these terms so they know their
-rights.
+- Timothy Lottes CRT-styled scan-line GLSL shader, in file
+  contrib/resources/glshaders/crt-fakelottes-flat.glsl, is
+  licensed under the Public Domain.
```
[the remark this answers is not in the document]
Comment (kcgen) [reply COPYING:26]: thanks, is this better wording?

Timothy Lottes' has dedicated his CRT-styled scan-line GLSL shader, in file `contrib/resources/glshaders/crt-fakelottes-flat.glsl`, to the public domain.

If so, then I'll include `Unlicense.txt` and/or `CC0.txt` in the licenses/ directory, and should I make reference to either of them in this here, too?

Timothy Lottes' has dedicated his CRT-styled scan-line GLSL shader, in file `contrib/resources/glshaders/crt-fakelottes-flat.glsl`, to the public domain (See the "Unlicense" and/or "CC0" licenses).

[the remark this answers is not in the document]
Comment (GranMinigun) [reply COPYING:26]: Header mentions [CC-PDDC](https://spdx.org/licenses/CC-PDDC.html).

**Comment by kcgen** [comment]
All comments should be addressed. Ready for round two.

**Comment by kcgen** [comment]
Thanks for help with this, @shermp and @GranMinigun.

**Pull request merged**